OneSignalSDK/onesignal/src/main/java/com/onesignal/OSInAppMessageRepository.java

```diff
@@ -15,12 +16,40 @@
 
 class OSInAppMessageRepository {
 
+    private static final long SIX_MONTHS_TIME_SECONDS = 6 * 30 * 24 * 60 * 60;
     private final OneSignalDbHelper dbHelper;
 
     OSInAppMessageRepository(OneSignalDbHelper dbHelper) {
         this.dbHelper = dbHelper;
     }
 
+    /**
+     * Remove IAMs that the last display time was six month ago
+     */
+    @WorkerThread
+    void deleteOldInAppMessages(long dateInSeconds) {
+        SQLiteDatabase readableDb = dbHelper.getReadableDbWithRetries();
```

Comment:
We could use a writable db here and performance a single write. Instead getting a list of ids then doing a delete on each record one at a time.

Reply:
Absolutely right